File: app/src/main/java/com/sd/stream/TestView.java
package com.sd.stream;

import android.content.Context;
import android.util.AttributeSet;
import android.widget.FrameLayout;

import com.sd.lib.stream.FStream;
import com.sd.lib.stream.FStreamManager;

public class TestView extends FrameLayout
{
    public TestView(Context context, AttributeSet attrs)
    {
        super(context, attrs);

        // 绑定流对象到当前View，会自动注册和取消注册
        FStreamManager.getInstance().bindStream(mFragmentCallback, this);
    }

    private final TestFragment.FragmentCallback mFragmentCallback = new TestFragment.FragmentCallback()
    {
        @Override
        public String getDisplayContent()
        {
            return "TestView";
        }

        @Override
        public Object getTagForStream(Class<? extends FStream> clazz)
        {
            return null;
        }
    };
}

File: app/src/main/java/com/sd/demo/stream/TestView.kt
package com.sd.demo.stream

import android.content.Context
import android.util.AttributeSet
import android.util.Log
import android.widget.FrameLayout
import com.sd.lib.stream.FStream
import com.sd.lib.stream.FStreamManager

class TestView : FrameLayout {
    constructor(context: Context, attrs: AttributeSet?) : super(context, attrs)

    private val _callback: TestFragment.FragmentCallback = object : TestFragment.FragmentCallback {
        override fun getDisplayContent(): String {
            Log.i(TestView::class.java.simpleName, "getDisplayContent")
            return "TestView"
        }

        override fun getTagForStream(clazz: Class<out FStream?>): Any? {
            return null
        }
    }

    override fun onAttachedToWindow() {
        super.onAttachedToWindow()
        FStreamManager.bindStream(_callback, this)
    }
}